celltype_annotation/utils/cellassign-helper-functions.R

```diff
@@ -72,3 +72,188 @@ compare_refs_heatmap <- function(original_assignment,
                            main = title)
   
 }
+
+# function to calculate the median delta for each cellassign result
+# input is full table of predictions returned from running cellassign
+get_median_delta_cellassign <- function(cellassign_predictions){
+  
+  # grab all the scores
+  preds <- cellassign_predictions |>
+    dplyr::select(-barcode) |>
+    as.matrix()
+  
+  # calculate the median delta (max score - median of all scores) 
+  median_delta <- rowMaxs(preds) - rowMedians(preds)
+  
+  # join the median delta with barcode 
+  median_delta_df <- data.frame(
+    barcode = cellassign_predictions$barcode,
+    median_delta = median_delta
+    )
+  
+  return(median_delta_df)
+}
+
+# create plot to look at median delta
+# input is combined data frame with assigned celltype and all delta median scores
+# also indicate what to color points by, default is celltype 
+# input dataframe must contain `median_delta` column to color points by 
+plot_median_delta <- function(celltype_results,
+                              color_group = "celltype"){
+  
+  # sina plot of delta median score for cellassign
+  # color_group is on the x-axis and used for calculating stats 
+  delta_plot <- ggplot(celltype_results, aes(x = !!rlang::sym(color_group), y = median_delta)) +
```

Comment:
Just noting (and I think I'm right this time 😉 ) that if you make this argument _not a string_, you can use `{{}}` as in `aes(x = {{color_group}})`. 
This is a small general comment though, no need to overhaul the code for something like this. If/when we move things over to `scpca-nf`, we can make more of those kinds of decisions then.

Reply:
I swear I actually tried that first and it wasn't working... but I can try again!